ethereum/spec/src/main/java/tech/pegasys/teku/spec/util/BeaconStateUtil.java

```diff
@@ -13,14 +13,66 @@
 
 package tech.pegasys.teku.spec.util;
 
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Math.toIntExact;
+import static tech.pegasys.teku.datastructures.util.ValidatorsUtil.get_active_validator_indices;
+import static tech.pegasys.teku.datastructures.util.ValidatorsUtil.increase_balance;
+
+import java.nio.ByteOrder;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.crypto.Hash;
+import tech.pegasys.teku.bls.BLS;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.datastructures.blocks.BeaconBlock;
+import tech.pegasys.teku.datastructures.operations.Deposit;
+import tech.pegasys.teku.datastructures.operations.DepositData;
+import tech.pegasys.teku.datastructures.operations.DepositMessage;
+import tech.pegasys.teku.datastructures.operations.DepositWithIndex;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+import tech.pegasys.teku.datastructures.state.BeaconStateCache;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.datastructures.state.ForkData;
+import tech.pegasys.teku.datastructures.state.MutableBeaconState;
+import tech.pegasys.teku.datastructures.state.SigningData;
+import tech.pegasys.teku.datastructures.state.Validator;
+import tech.pegasys.teku.datastructures.util.GenesisGenerator;
+import tech.pegasys.teku.datastructures.util.ValidatorsUtil;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.ssz.SSZTypes.Bitvector;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+import tech.pegasys.teku.ssz.SSZTypes.SSZList;
+import tech.pegasys.teku.ssz.SSZTypes.SSZVector;
+import tech.pegasys.teku.ssz.backing.Merkleizable;
+import tech.pegasys.teku.ssz.backing.schema.SszComplexSchemas;
+import tech.pegasys.teku.ssz.backing.view.SszPrimitives;
 
+@SuppressWarnings("unused")
 public class BeaconStateUtil {
+  private static final Logger LOG =
+      LogManager.getLogger(tech.pegasys.teku.datastructures.util.BeaconStateUtil.class);
+  /**
+   * For debug/test purposes only enables/disables {@link DepositData} BLS signature verification
+   * Setting to <code>false</code> significantly speeds up state initialization
+   */
+  public final boolean BLS_VERIFY_DEPOSIT = true;
+
   private final SpecConstants specConstants;
+  private final Spec spec;
 
-  public BeaconStateUtil(final SpecConstants specConstants) {
-    this.specConstants = specConstants;
+  public BeaconStateUtil(final Spec spec) {
+    this.spec = spec;
```

Comment:
Instead of injecting the full `spec`, we should probably just inject the specific utils we need.  This avoids circular dependencies and makes the actual dependency requirements clearer.